Sources/Build/BuildPlan/BuildPlan+Swift.swift

```diff
@@ -26,6 +26,10 @@ extension BuildPlan {
                 guard case let .clang(target)? = description else {
                     throw InternalError("unexpected clang target \(underlyingTarget)")
                 }
+                // Only if the targets are for the same destination, i.e. not a plugin for a target target
+                guard target.buildParameters.destination == swiftTarget.buildParameters.destination else {
+                    break
+                }
```

Comment:
formatting nit: IMO more readable with visual separation, and makes it easier to navigate with `{`/`}` when using Vi keybindings 😉 

```suggestion
                }

                // Only if the targets are for the same destination, i.e. not a plugin for a target target
                guard target.buildParameters.destination == swiftTarget.buildParameters.destination else {
                    break
                }

```